src/SignalR/clients/ts/signalr/src/HubConnectionBuilder.ts

```diff
@@ -138,8 +138,16 @@ export class HubConnectionBuilder {
         } else {
             this.httpConnectionOptions = {
                 ...this.httpConnectionOptions,
-                transport: transportTypeOrOptions,
             };
+
+            if(this.httpConnectionOptions.transport && transportTypeOrOptions) {
```

Comment:
```suggestion
            if (this.httpConnectionOptions.transport && !transportTypeOrOptions) {
```

Shouldn't this be an undefined check?